besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -1178,6 +1221,20 @@ private JsonRpcConfiguration jsonRpcConfiguration() {
           "Unable to authenticate JSON-RPC HTTP endpoint without a supplied credentials file or authentication public key file");
     }
 
+    if (isRpcHttpEnabled && isRpcHttpTlsEnabled) {
```

Comment:
If you move this conditional lower, you don't need to duplicate it (at line 1249).
I.e.create jsonRpcConfiguration before doing these checks.

Reply:
I still want to keep this check here before creating jsonRpcConfiguration, but I can create TLSConfiguration object here hence eliminating the need to if/else condition appearing twice.

Reply:
done